realm/realm-library/src/main/cpp/objectserver_shared.hpp

```diff
@@ -81,6 +82,9 @@ class JniSession {
                             local_env, obj, "notifySessionError", "(ILjava/lang/String;)V");
                     local_env->CallVoidMethod(
                             obj, notify_error_handler, error_code.value(), local_env->NewStringUTF(message.c_str()));
+                    if (local_env->ExceptionCheck() == JNI_TRUE) {
+                        throw realm::jni_util::JniPendingException("Exception in error handler");
```

Comment:
I think that needs to be handled separately. For current implementation (SyncManager in the java side), if it is not a fatal error, we need to quit the client native loop, throw the exception, then call the `client::run` again. (But it needs to be aligned with sync.)

But do we want to throw an java exception in the client thread if it is not a fatal error? Isn't it good enough just pass the error to the client error handler?